[ADD] default_warehouse_from_sale_team: New field pricelist_team_ids.

ysantiago opened the pull request:
- Added field m2m pricelist_team_ids in Sale Team to manage access users to pricelist objects (product.pricelist)
- Added res_groups and ir_rule to manage access to pricelists in Sale Teams.

Close [Vauxoo/typ#39](https://github.com/Vauxoo/typ/pull/39)
Close [Vauxoo/yoytec#2003](https://github.com/Vauxoo/yoytec/pull/2003)

ysantiago committed:
[ADD] default_warehouse_from_sale_team: New field m2m pricelist_team_ids.
[ADD] default_warehouse_from_sale_team: New res_groups and ir_rule to manage access to pricelists in sale teams.

ysantiago committed:
[ADD] default_warehouse_from_sale_team: Add tests.

ysantiago commented on default_warehouse_from_sale_team/__openerp__.py:
```diff
@@ -15,6 +15,7 @@
         'sales_team',
         'base_action_rule',
         'purchase_requisition',
+        'warning',
```
@JulioSerna I have to add this depend [odoo/addons/warning](https://github.com/odoo/odoo/tree/9.0/addons/warning) because in test when is created the new user this error appears:

<img width="1278" alt="skitch_13" src="https://cloud.githubusercontent.com/assets/11341058/18555830/466f3ad4-7b37-11e6-9555-300ed2617a0d.png">

JulioSerna replied:
@ysantiago 
this tests was tested only installing this module?

I think we do not need to depend to that module, maybe you can test using another user demo

ysantiago replied:
@JulioSerna I probed using Demo User and tests doesn`t run properly.

ysantiago replied:
The dependence to website module was remove in [9e00310](https://github.com/Vauxoo/addons-vauxoo/pull/1060/commits/9e003100c158aac6f2d69b8f6ac6434e17b39443)

ysantiago committed:
[REF] default_warehouse_from_sale_team: Remove dependence to warning module.

ysantiago committed:
[ADD] defualt_warehouse_from_sale_team: Adding new field default_sale_pricelist and default_purchase_pricelist.

ysantiago committed:
[ADD] defualt_warehouse_from_sale_team: Adding test to new fields default_sale_pricelist and default_purchase_pricelist.

JulioSerna commented:
@ysantiago 
1. search the sale team by warehouse and not by te user
2. search first the `pricelist_id` into sale team by `default_purchase_pricelist/default_sale_pricelist` fields if not value search into `pricelist_team_ids` fields, if there is `pricelist_team_ids` you must validate if is sale or purchase the priceslist 
3. if not `pricelist_team_ids` or `default_purchase_pricelist/default_sale_pricelist` **you must return the original pricelist_id by the onchange**

ysantiago committed:
[REF] delivery_warehouse_from_sale_team: Inherit onchange_partner_id and onchange_warehouse_id to sale order to obtain the pricelist by default.
[REF] delivery_warehouse_from_sale_team: Inherit onchage_partner_id and onchange_picking_type_id to purchase_order to obtain pricelist by default.
[REF] delivery_warehouse_from_sale_team: Extend view sale_order form and purchase_order form to change onchage_partner_id.
[ADD] delivery_warehouse_from_sale_team: Adding constraint in sale_team to default_sale_pricelist and default_purchase_pricelist fields.

ysantiago committed:
[REF] delivery_warehouse_from_sale_team: Adapt tests to changes.

ysantiago commented:
@JulioSerna can you please review the changes that you suggested?

ysantiago committed:
[REF] default_warehouse_from_sale_team: In test give access level Purchase/user to the new user test.

ysantiago committed:
[REF] default_warehouse_from_sale_team: Adapt translation.

ysantiago committed:
[REF] default_warehouse_from_sale_team: Add domains in fields default_sale_pricelist and default_purchase_pricelist.

ysantiago committed:
[REF] default_warehouse_from_sale_team: Change onchange's to new api.

ysantiago committed:
[REF] delivery_warehouse_from_sale_team: Adapt tests to changes.

ysantiago commented:
This PR is going to be closed by changes in the development guidelines.

Changes was made in [Vauxoo/typ#43](https://github.com/Vauxoo/typ/pull/43)
